listener/carbon_plaintext.go

```diff
@@ -176,6 +177,9 @@ func metricHandler(line string) {
 		return
 	}
 
+	// Send metric path off to be indexed.
+	go datastore.IndexMetricPath(statPath)
```

Comment:
This will be done after the metric is sent down the channel, by the code that reads it out of the channel.

I currently have this as QueueManager, but I'm coming up with a better abstraction there.